app/src/main/kotlin/com/captures2024/soongan/state/AppRootUIState.kt

```diff
@@ -0,0 +1,26 @@
+package com.captures2024.soongan.state
+
+import com.captures2024.soongan.core.analytics.utils.LogElementArgument
+import com.captures2024.soongan.core.common.base.UIState
+import com.captures2024.soongan.core.model.dto.UserInfoDto
+
+internal data class AppRootUIState(
+    val isLoading: Boolean = false,
+    val rootRouteState: AppRootRouteState = AppRootRouteState.LANDING,
+    private val fcmToken: String = "",
+    private val accessToken: String = "",
+    private val refreshToken: String = "",
+    private val memberInfo: UserInfoDto = UserInfoDto.defaultBuilder(),
+) : UIState {
```

Comment:
```suggestion
internal data class AppRootUIState(
    val isLoading: Boolean = false,
    val rootRouteState: AppRootRouteState = AppRootRouteState.LANDING,
    val isGuestMode: Boolean = false,
) : UIState {
```
private이긴 하지만 token과 memberInfo가 UIState에 필수적인 값이 아닌 것 같아 제안드려 봅니다.

Reply:
app root ui state를 구현할 때

제가 고려했던 부분은 앱 전체에서 user info를 관리하고 있는 vm이 존재해야한다고 생각해서 appRoot vm을 만들고 거기에서 관찰할 수 있는 State를 만들고자했던 부분이었습니다

ui를 그리는데 필요한 상태값이라는 전제에서 각각 token과 user info를 vm 한곳에서만 관리한다면 member info가 없을 때는 routeState를 Sign으로 전환 시켜서 로그인이 필요하게 하고 할 수 있는 작업을 할 수 있을 것 같아서 관리했던 부분인데

더 좋은 방법이 있거나 하면 변경해도 됩니다

제가 생각했을 땐 전체적으로 유저 정보를 가지고 있는 vm이 필요하긴 한 것 같아서요

Reply:
token은 DataStore에 존재해서 필요한 화면이 있더라도 해당 vm에서 읽어 사용하는데 비용이 크지 않다고 생각했어요.

memberInfo는 저도 필요한 부분이 있을까 생각을 해보았는데 현재는 게스트 모드로 접근하지 못하는 화면에서 로그인으로 전환해주는 경우만 있다고 생각이 들어서 작성했어요.

모든 기능이 구체적으로 정의된 것은 아니니 진행하시고 이후 고려해보는게 나을 것 같네요.

Reply:
토큰은 의견 이해했습니다

테스트 해보고 삭제할 수 있으면 삭제할 수 있게 해볼게요